<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Camera</class>
 <widget class="QDialog" name="Camera">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>570</width>
    <height>413</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>相机设置</string>
  </property>
  <property name="windowIcon">
   <iconset resource="detection.qrc">
    <normaloff>:/Detection/Resources/14.png</normaloff>:/Detection/Resources/14.png</iconset>
  </property>
  <widget class="QGroupBox" name="groupBox">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>20</y>
     <width>211</width>
     <height>211</height>
    </rect>
   </property>
   <property name="title">
    <string>相机参数</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
   </property>
   <widget class="QComboBox" name="comboBox">
    <property name="geometry">
     <rect>
      <x>80</x>
      <y>20</y>
      <width>121</width>
      <height>21</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>20</y>
      <width>61</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>选择相机：</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>50</y>
      <width>61</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>相机模型：</string>
    </property>
   </widget>
   <widget class="QComboBox" name="comboBox_2">
    <property name="geometry">
     <rect>
      <x>80</x>
      <y>49</y>
      <width>86</width>
      <height>20</height>
     </rect>
    </property>
    <property name="editable">
     <bool>false</bool>
    </property>
    <property name="sizeAdjustPolicy">
     <enum>QComboBox::AdjustToContents</enum>
    </property>
    <item>
     <property name="text">
      <string>Polynomial</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Division</string>
     </property>
    </item>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>90</y>
      <width>81</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>单个像元的高：</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_9">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>120</y>
      <width>91</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>单个像元的宽：</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_10">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>150</y>
      <width>61</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>焦距：</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_5">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>90</y>
      <width>61</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_6">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>120</y>
      <width>61</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_7">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>150</y>
      <width>61</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_11">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>90</y>
      <width>21</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>μm</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_12">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>120</y>
      <width>21</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>μm</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_13">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>150</y>
      <width>21</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>mm</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_7">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>180</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>导入参数</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_8">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>180</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>打开相机</string>
    </property>
   </widget>
  </widget>
  <widget class="QGroupBox" name="groupBox_2">
   <property name="geometry">
    <rect>
     <x>230</x>
     <y>20</y>
     <width>331</width>
     <height>351</height>
    </rect>
   </property>
   <property name="title">
    <string>相机标定</string>
   </property>
   <widget class="QPushButton" name="pushButton_2">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>20</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>采集图片</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_3">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>20</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>导入图片</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_4">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>20</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>删除图片</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_5">
    <property name="geometry">
     <rect>
      <x>250</x>
      <y>20</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>开始标定</string>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>50</y>
      <width>71</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>已选择图片</string>
    </property>
   </widget>
   <widget class="QListWidget" name="listWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>70</y>
      <width>311</width>
      <height>271</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>1</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>2</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>3</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>4</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>5</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>6</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>7</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>8</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>9</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>10</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>11</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>12</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>13</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>14</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>15</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>16</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>17</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>18</string>
     </property>
    </item>
   </widget>
  </widget>
  <widget class="QProgressBar" name="progressBar">
   <property name="geometry">
    <rect>
     <x>230</x>
     <y>380</y>
     <width>331</width>
     <height>23</height>
    </rect>
   </property>
   <property name="value">
    <number>24</number>
   </property>
  </widget>
  <widget class="QGroupBox" name="groupBox_3">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>240</y>
     <width>211</width>
     <height>171</height>
    </rect>
   </property>
   <property name="title">
    <string>标定板</string>
   </property>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>20</y>
      <width>101</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>X方向标志点数量：</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>50</y>
      <width>101</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Y方向标志点数量：</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>80</y>
      <width>131</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>标志点距离(mm)：</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>110</y>
      <width>131</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>标志点直径与距离比值：</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_6">
    <property name="geometry">
     <rect>
      <x>60</x>
      <y>140</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>导入参数</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>20</y>
      <width>61</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_2">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>50</y>
      <width>61</width>
      <height>21</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_3">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>80</y>
      <width>61</width>
      <height>21</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_4">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>110</y>
      <width>61</width>
      <height>21</height>
     </rect>
    </property>
   </widget>
  </widget>
 </widget>
 <resources>
  <include location="detection.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>comboBox</sender>
   <signal>activated(QString)</signal>
   <receiver>Camera</receiver>
   <slot>update_device()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>150</x>
     <y>50</y>
    </hint>
    <hint type="destinationlabel">
     <x>284</x>
     <y>206</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_5</sender>
   <signal>clicked()</signal>
   <receiver>Camera</receiver>
   <slot>calibrate_cameras()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>513</x>
     <y>49</y>
    </hint>
    <hint type="destinationlabel">
     <x>482</x>
     <y>10</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comboBox</sender>
   <signal>activated(int)</signal>
   <receiver>Camera</receiver>
   <slot>selectdevice(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>108</x>
     <y>48</y>
    </hint>
    <hint type="destinationlabel">
     <x>130</x>
     <y>13</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lineEdit_5</sender>
   <signal>textChanged(QString)</signal>
   <receiver>Camera</receiver>
   <slot>addparameters()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>161</x>
     <y>115</y>
    </hint>
    <hint type="destinationlabel">
     <x>274</x>
     <y>10</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lineEdit_6</sender>
   <signal>textChanged(QString)</signal>
   <receiver>Camera</receiver>
   <slot>addparameters()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>124</x>
     <y>152</y>
    </hint>
    <hint type="destinationlabel">
     <x>258</x>
     <y>5</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lineEdit_7</sender>
   <signal>textChanged(QString)</signal>
   <receiver>Camera</receiver>
   <slot>addparameters()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>143</x>
     <y>177</y>
    </hint>
    <hint type="destinationlabel">
     <x>95</x>
     <y>12</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_7</sender>
   <signal>clicked()</signal>
   <receiver>Camera</receiver>
   <slot>importdata()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>59</x>
     <y>209</y>
    </hint>
    <hint type="destinationlabel">
     <x>5</x>
     <y>78</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_8</sender>
   <signal>clicked()</signal>
   <receiver>Camera</receiver>
   <slot>startcamera()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>200</x>
     <y>213</y>
    </hint>
    <hint type="destinationlabel">
     <x>223</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comboBox_2</sender>
   <signal>activated(int)</signal>
   <receiver>Camera</receiver>
   <slot>changemode(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>175</x>
     <y>75</y>
    </hint>
    <hint type="destinationlabel">
     <x>225</x>
     <y>27</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lineEdit</sender>
   <signal>textChanged(QString)</signal>
   <receiver>Camera</receiver>
   <slot>addPlatedata()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>191</x>
     <y>261</y>
    </hint>
    <hint type="destinationlabel">
     <x>224</x>
     <y>233</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lineEdit_2</sender>
   <signal>textChanged(QString)</signal>
   <receiver>Camera</receiver>
   <slot>addPlatedata()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>202</x>
     <y>297</y>
    </hint>
    <hint type="destinationlabel">
     <x>224</x>
     <y>291</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lineEdit_3</sender>
   <signal>textChanged(QString)</signal>
   <receiver>Camera</receiver>
   <slot>addPlatedata()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>165</x>
     <y>328</y>
    </hint>
    <hint type="destinationlabel">
     <x>226</x>
     <y>324</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lineEdit_4</sender>
   <signal>textChanged(QString)</signal>
   <receiver>Camera</receiver>
   <slot>addPlatedata()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>170</x>
     <y>357</y>
    </hint>
    <hint type="destinationlabel">
     <x>239</x>
     <y>373</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_6</sender>
   <signal>clicked()</signal>
   <receiver>Camera</receiver>
   <slot>importPlatedata()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>119</x>
     <y>390</y>
    </hint>
    <hint type="destinationlabel">
     <x>152</x>
     <y>397</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>Camera</receiver>
   <slot>capture()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>306</x>
     <y>47</y>
    </hint>
    <hint type="destinationlabel">
     <x>311</x>
     <y>12</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_3</sender>
   <signal>clicked()</signal>
   <receiver>Camera</receiver>
   <slot>addimage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>371</x>
     <y>51</y>
    </hint>
    <hint type="destinationlabel">
     <x>393</x>
     <y>13</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_4</sender>
   <signal>clicked()</signal>
   <receiver>Camera</receiver>
   <slot>deletiamage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>448</x>
     <y>45</y>
    </hint>
    <hint type="destinationlabel">
     <x>468</x>
     <y>13</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_4</sender>
   <signal>clicked()</signal>
   <receiver>Camera</receiver>
   <slot>deletimage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>423</x>
     <y>53</y>
    </hint>
    <hint type="destinationlabel">
     <x>392</x>
     <y>2</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <signal>cameraStart()</signal>
  <signal>cameraStop()</signal>
  <slot>update_device()</slot>
  <slot>calibrate_cameras()</slot>
  <slot>selectdevice(int)</slot>
  <slot>startcamera()</slot>
  <slot>importdata()</slot>
  <slot>addparameters()</slot>
  <slot>changemode(int)</slot>
  <slot>importPlatedata()</slot>
  <slot>addPlatedata()</slot>
  <slot>capture()</slot>
  <slot>addimage()</slot>
  <slot>deletimage()</slot>
 </slots>
</ui>
